### PR TITLE
rename storageclassclaim to storageclaim in gRPC methods

### PR DESCRIPTION
**Pull request by leelavg**
Depends on red-hat-storage/ocs-operator#2548

### CODE REVIEW

**Comment by leelavg** [comment]
/hold

for dependencies & custom go.mod replace.

File: go.mod [hunk 1]
```diff
@@ -2,6 +2,8 @@ module github.com/red-hat-storage/ocs-client-operator
 
 go 1.21
 
+replace github.com/red-hat-storage/ocs-operator/v4 => github.com/leelavg/ocs-operator/v4 v4.0.0-20240408073504-3ac8570eef74
```
Comment (nb-ohad) [inline go.mod:5]: This need to go away before merging

File: controllers/storageclaim_controller.go [hunk 2]
```diff
@@ -286,15 +286,15 @@ func (r *StorageClaimReconciler) reconcilePhases() (reconcile.Result, error) {
 		var storageClaimStorageType providerclient.StorageType
 		switch r.storageClaim.Spec.Type {
 		case "block":
-			storageClaimStorageType = providerclient.StorageTypeBlockpool
+			storageClaimStorageType = providerclient.StorageTypeBlock
 		case "sharedfile":
-			storageClaimStorageType = providerclient.StorageTypeSharedfilesystem
+			storageClaimStorageType = providerclient.StorageTypeSharedfile
```
Comment (nb-ohad) [inline controllers/storageclaim_controller.go:291]:
```suggestion
			storageClaimStorageType = providerclient.StorageTypeSharedFile
```

**Review by nb-ohad — CHANGES_REQUESTED** [review]

File: go.mod (hunk 1 above)
Comment (leelavg) [reply go.mod]: done.

File: controllers/storageclaim_controller.go (hunk 2 above)
Comment (leelavg) [reply controllers/storageclaim_controller.go:280]: will be taken up in a different PR to cut across all the components and change to `SharedFile` and `Block` as the values.

**Pull request merged**